TezosSwift/Client/TezosClient.swift

```diff
@@ -545,7 +591,7 @@ public class TezosClient {
 
         if method == .post {
             do {
-                guard let payload = payload else { return }
+				guard let payload = payload else { return nil } // TODO: maybe call completion with .encryptionFailed
```

Comment:
What exactly do you mean by this TODO?

Reply:
In this case completion will never be called and could result in an unexpected behavior. User completion should always be called no matter what happen.

Reply:
I agree, so could you please resolve this? I'd copy the `case decryptionFailed(reason: DecryptionReason)` from `TezosError` and implement it for encryption.